orttraining/orttraining/models/runner/training_runner.cc

```diff
@@ -34,6 +34,7 @@ namespace training {
 static std::vector<FreeDimensionOverride> overrides = {};
 static SessionOptions SESSION_OPTION = {
     ExecutionMode::ORT_SEQUENTIAL,     //execution_mode
+    ExecutionOrder::PRIORITY_BASED,    //execution_order
```

Comment:
> ExecutionOrder::PRIORITY_BASED [](start = 3, length = 31)

Should we use it only when enabling transformer layer recompute? It's better to not change baseline.

Reply:
C++ frontend is just for development purpose... I feel it's probably fine to leave it as it is. 
I also tried the E2E test for bert, and it actually didn't' affect the convergence.